cmd/flakechecker/bq_client.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"cloud.google.com/go/bigquery"
+	"context"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/api/iterator"
+	"regexp"
+	"time"
+)
+
+const projectID = "acs-san-stackroxci"
+const queryTimeout = 1 * time.Minute
+
+type biqQueryClient interface {
+	GetRatioForTest(flakeTestConfig *flakeCheckerRecord, testName string) (int, int, error)
+}
+
+type biqQueryClientImpl struct {
+	client *bigquery.Client
+}
+
+func getNewBigQueryClient() (biqQueryClient, error) {
+	ctx := context.Background()
+
+	client, err := bigquery.NewClient(ctx, projectID)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating BigQuery client")
+	}
+
+	return &biqQueryClientImpl{client: client}, nil
+}
+
+func getFilteredTestName(name string) string {
+	// Apply the same filtering used in DB to generate test names not influenced with version Z number.
+	regFilterTestName := regexp.MustCompile(`(.*)((-v|: )\d{3}\.\d+\.)(\d+)(.*)`)
+	return regFilterTestName.ReplaceAllString(name, `${1}${2}z${5}`)
+}
+
+func (c *biqQueryClientImpl) GetRatioForTest(flakeTestConfig *flakeCheckerRecord, testName string) (int, int, error) {
+	query := c.client.Query(
+		`SELECT JobName, FilteredName, Classname, TotalAll, FailRatio
+    FROM acs-san-stackroxci.ci_metrics.stackrox_tests_recent_flaky_tests
+    WHERE JobName = @jobName AND FilteredName = @filteredName AND Classname = @classname
+`)
+
+	query.Parameters = []bigquery.QueryParameter{
+		{Name: "jobName", Value: flakeTestConfig.RatioJobName},
+		{Name: "filteredName", Value: getFilteredTestName(testName)},
+		{Name: "classname", Value: flakeTestConfig.Classname},
+	}
+
+	ctx, cancelBigQueryRequest := context.WithTimeout(context.Background(), queryTimeout)
+	defer cancelBigQueryRequest()
+
+	resIter, err := query.Read(ctx)
+	if err != nil {
+		return 0, 0, errors.Wrap(err, "query data from BigQuery")
+	}
+
+	// We need only first record. No need to loop over iterator.
```

Comment:
If we need only the first record how about ensuring that query will return single value?

Reply:
After reading error in 69 this comment is not true (or maybe it's softer then it could be). We do require single row